compiler/lib/php_from_rehp.re

```diff
@@ -278,21 +279,35 @@ let topLevelIdentifiersSt = (newVarsSoFar, st) =>
   | _ => newVarsSoFar
   };
 
-let topLevelIdentifiers = (newVarsSoFar, (src, _)) =>
+let topLevelIdentifiers = (newVarsSoFar: vars, (src, _)) =>
   switch (src) {
   | Rehp.Function_declaration((id, _, _, _)) => addOne(newVarsSoFar, id)
   | Statement(stmt) => topLevelIdentifiersSt(newVarsSoFar, stmt)
   };
 
-let rec foldSources = (sourceFolder, curOut, curIn, curRevMappeds, remain) =>
+let rec foldSources =
+        (
+          sourceFolder,
+          origOut: output,
+          curOut: output,
+          curIn: input,
+          curRevMappeds,
+          remain,
+        ) =>
   switch (remain) {
   | [] => (curOut, List.rev(curRevMappeds))
   | [(s, loc), ...tl] =>
-    let (thisOut, thisMapped) = sourceFolder(curOut, curIn, s);
+    let (thisOut, thisMapped) =
+      sourceFolder(
+        {...curOut, unshielded_continues: origOut.unshielded_continues},
+        curIn,
+        s,
+      );
     let nextOut = outAppend(curOut, thisOut);
-    let nextInput = append(curIn, nextOut.dec);
+    let nextInput = {vars: append(curIn.vars, nextOut.dec), label: NoLabel};
```

Comment:
this seems correct - each statement in a function body has a reset "scope" of labels (unlike variable scope which inherits the function's scope, and appends across subsequent statements in a function body).